### PR TITLE
2.3 only - creating attribute option value using API returns unexpected response #18392

### PR DESCRIPTION
**Pull request by Nazar65**
### Description (*)
Return value should be the ID of the new option

### Fixed Issues (if relevant)

1. magento/magento2#18392: 2.3 only - creating attribute option value using API returns unexpected response

### Manual testing scenarios (*)

1. Setup a clean Magento installation
2. Create an Integration with full access rights & activate it, I'm assuming the access token is abcd (see step 4)
3. Using the adminhtml, create a new dropdown product attribute, use attribute code: dropdown
4. Add an option through a REST call (make sure to use an option which doesn't exist yet):
`curl -k -X POST "https://domain.tld/rest/all/V1/products/attributes/dropdown/options" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer abcd" -d "{ \"option\": { \"label\": \"whatever\" }}`
5. Value should be the ID of the new option

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-3413](https://jira.corp.magento.com/browse/ENGCOM-3413) has been created to process this Pull Request

File: app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php [hunk 1]
```diff
@@ -6,36 +6,49 @@
...
+     * @param Attribute $resourceModel
+     * @param GetAttributeOptionId $getAttributeOptionId
      * @codeCoverageIgnore
      */
     public function __construct(
-        \Magento\Eav\Model\AttributeRepository $attributeRepository,
-        \Magento\Eav\Model\ResourceModel\Entity\Attribute $resourceModel
+        AttributeRepository $attributeRepository,
+        Attribute $resourceModel,
+        GetAttributeOptionId $getAttributeOptionId
```
Comment (sidolov) [inline app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php (hunk 1 above)
Comment (Nazar65) [reply app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]: ok

**Comment by Nazar65** [comment]
Hi @sidolov all done.

File: app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php [hunk 2]
```diff
@@ -14,6 +14,7 @@
 use Magento\Framework\Exception\InputException;
 use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Framework\Exception\StateException;
+use \Magento\Framework\App\ObjectManager;
```
Comment (hostep) [inline app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]: The first `\` can be removed here so it's consistent with the use statements above it.

Comment (Nazar65) [reply app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]: @hostep Thanks for some reason I did not see it right away

**Comment by Nazar65** [comment]
after some changes in 2.3-develop now we have "returned id of option". So no need anymore this PR. Created at on Nov 7, 2018 :confused: 
screen

**Pull request closed without merging**